modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AlterTableDropColumnCommand.java

```diff
@@ -98,24 +96,39 @@ public List<UpdateEntry> get(Catalog catalog) {
             }
 
             if (indexedColumns.contains(columnName)) {
-                List<String> indexesNames = Arrays.stream(schema.indexes())
-                        .filter(index -> index.tableId() == table.id())
-                        .filter(index -> index instanceof CatalogHashIndexDescriptor
-                                ? ((CatalogHashIndexDescriptor) index).columns().contains(columnName)
-                                : ((CatalogSortedIndexDescriptor) index).columns().stream().map(CatalogIndexColumnDescriptor::name)
-                                        .anyMatch(column -> column.equals(columnName))
-                ).map(CatalogIndexDescriptor::name).collect(Collectors.toList());
+                List<String> indexesNames = indexesForTable(schema, table)
+                        .filter(index -> indexColumnNames(index).anyMatch(columnName::equals))
+                        .map(CatalogIndexDescriptor::name)
+                        .collect(Collectors.toList());
 
                 throw new CatalogValidationException(format(
                         "Deleting column '{}' used by index(es) {}, it is not allowed", columnName, indexesNames));
             }
-        }
+        });
 
         return List.of(
                 new DropColumnsEntry(table.id(), columns, schemaName)
         );
     }
 
+    private static Stream<CatalogIndexDescriptor> indexesForTable(CatalogSchemaDescriptor schema, CatalogTableDescriptor table) {
+        return Arrays.stream(schema.indexes())
+                .filter(index -> index.tableId() == table.id() && index.status() != CatalogIndexStatus.STOPPING);
```

Comment:
What will this method do? Simply compare its status to `STOPPING`?

Reply:
Yes. The idea is that in the code using this method it will be clear that 'we want an alive index' and not that 'we want a not STOPPED index'; the latter seems to be a little too low-level, the former explains our intention.

Reply:
I'm not insisting, this is a little thing anyway, but adding an instance method seems easy, like

```
    public boolean alive() {
        return this == STOPPING;
    }
```

Reply:
Sorry, I got stupid and thought that I would need to add an abstract method. Will do then.

Reply:
Fixed